sdk/core/azure-core/test/ut/service_tracing_test.cpp

```diff
@@ -32,6 +32,82 @@ TEST(TracingContextFactory, ServiceTraceEnums)
   std::string tracingAttributeName = TracingAttributes::AzNamespace.ToString();
 }
 
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <azure/core/http/policies/policy.hpp>
+#include <azure/core/internal/http/pipeline.hpp>
+#include <gtest/gtest.h>
+
+using namespace Azure::Core;
+using namespace Azure::Core::Http;
+using namespace Azure::Core::Http::_internal;
+using namespace Azure::Core::Http::Policies;
+using namespace Azure::Core::Http::Policies::_internal;
+
+namespace {
+
+class NoOpPolicy final : public HttpPolicy {
+private:
+  std::unique_ptr<RawResponse> Send(
+      Request& request,
+      NextHttpPolicy nextPolicy,
+      Context const& context) const override
+  {
+    (void)context;
+    (void)request;
+    (void)nextPolicy;
+
+    return std::unique_ptr<RawResponse>();
+  }
+
+  std::unique_ptr<HttpPolicy> Clone() const override { return std::make_unique<NoOpPolicy>(*this); }
+};
+
+} // namespace
+
+TEST(TracingContextFactory, UserAgentTests)
+{
+  struct
+  {
+    const std::string serviceName;
+    const std::string serviceVersion;
+    const std::string applicationId;
+    const std::string expectedPrefix;
+  } UserAgentTests[]
+      = {{"storage-blob", "11.0.0", "", "azsdk-cpp-storage-blob/11.0.0 ("},
+         {"storage-blob",
+          "11.0.0",
+          "AzCopy/10.0.4-Preview",
+          "AzCopy/10.0.4-Preview azsdk-cpp-storage-blob/11.0.0 ("},
+         {"storage-blob",
+          "11.0.0",
+          "AzCopy / 10.0.4-Preview ",
+          "AzCopy / 10.0.4-Preview azsdk-cpp-storage-blob/11.0.0 ("},
+         {"storage-blob",
+          "11.0.0",
+          "  01234567890123456789abcde  ",
+          "01234567890123456789abcd azsdk-cpp-storage-blob/11.0.0 ("}};
+
+  constexpr auto UserAgentEnd = ')';
+  constexpr auto OSInfoMinLength = 10;
```

Comment:
This will currently fail on UWP, where we currently put just `"UWP"` (because we haven't figured out how to get OS info yet, and left it for the future)

Reply:
This is the existing test collateral, just framed in a slightly different way. I rewrote it to be more data driven, but it is exactly the same test collateral we already have in the TelemetryPolicy tests. I'm not comfortable with changing existing test collateral.